src/EnergyPlus/DataZoneEquipment.hh

```diff
@@ -164,6 +164,9 @@ namespace DataZoneEquipment {
         Num
     };
 
+    constexpr std::array<std::string_view, static_cast<int>(LoadDist::Num)> LoadDistNamesUC = {
+        "SEQUENTIALLOAD", "UNIFORMLOAD", "UNIFORMPLR", "SEQUENTIALUNIFORMPLR"};
+
```

Comment:
This should be in the cc since it isn't yet used elsewhere other than DataZoneEquipment.

Reply:
This also worked!

Reply:
*Re: `.cc` vs. `.hh`*. Not that strongly. I think if you leave it in the `.hh` you have to declare it `static` or else you will get a copy of the array for each time the `.hh` file is included.

*Re: BETTER_ENUM (or whatever)*. Was not familiar with this. Looks ... interesting. I'm not very good at reading template code, but the implementation looks okay I guess. A faster solution for case-insensitive matching would be nice (e.g., a separate upper-case names array) and looping via recursion is too cute by 10X, but whatever. Are different `BETTER_ENUM` enumerations different types (like `enum class`, i.e., type safe but have to be `static_cast<int>`) or are they all `int` (like `enum`, i.e., not type safe but don't have to be `static_cast<int>`) or are they different types that transparently derive from `int` (i.e., type safe *and* don't have to be `static_cast`<int>` as `enum class` should have been? I will look at `OPENSTUDIO_ENUM` also. Thanks for the references.

Reply:
I'll move it to the .cc, but for completeness, if it's in a header and declared `static constexpr`, each TU will still get a copy. What we probably want then is `inline constexpr`.

As far as the BETTER_ENUM thing, the best way is to ask gcc to expand it

```shell
mkdir test_better_enum && cd test_better_enum
wget https://raw.githubusercontent.com/aantron/better-enums/master/enum.h
echo "#include "enum.h" > test.cpp
echo "" >> test.cpp
echo "BETTER_ENUM(LoadDist, int, Invalid = -1, Sequential, Uniform, UniformPLR, SequentialUniformPLR, Num);" >> test.cpp
g++ -std=c++20 -O3 test.cpp -E -o expanded.cpp
```

see https://gist.github.com/jmarrec/8a28cf529742b9c5c603ec77696f86af

Reply:
That's a lot of code per enum (not talking about the included headers, just the stuff that is generated on a per enum basis at the bottom). You are actually getting multiple implementations of the enum, a highly functional and type-safe one in which each enum is implemented as its own class (not its own `enum class`, its own `class`!) and then a more normal and less typesafe one as a C enum. The former is cute, but overkill. The latter is fine if you are willing to give up type safety to the extent that it is even needed for enumerations. FWIW if we were willing to do that, we could replace all of our `enum class` with `enum` and we would be able to drop the `static_cast<int>` everywhere.

This is cute, but I would rather stick with the way we are doing things at least for now. If there is a compelling reason to change, we can do a pass later.